client/lib/query-args/index.ts

```diff
@@ -0,0 +1,59 @@
+import { getQueryArgs } from '@wordpress/url';
+import page from 'page';
+
+function getRelativeUrlWithParameters( queryArgs: object ): string {
+	const urlParams = new URLSearchParams( window.location.search );
+	let returnPath = window.location.pathname;
+
+	if ( queryArgs ) {
+		for ( const [ key, value ] of Object.entries( queryArgs ) ) {
+			if ( '' !== value ) {
+				urlParams.set( key, value );
+			} else {
+				urlParams.delete( key );
+			}
+		}
+
+		returnPath += '?' + urlParams.toString();
+	}
+
+	return returnPath;
+}
+
+function useQueryArgs() {
+	function setQueryArgs( queryArgs: object ) {
```

Comment:
can we define this function outside of the hook like `getRelativeUrlWithParameters` since it doesn't have any dependency inside the hook.